crml/nft/src/lib.rs

```diff
@@ -0,0 +1,651 @@
+/* Copyright 2019-2021 Centrality Investments Limited
+*
+* Licensed under the LGPL, Version 3.0 (the "License");
+* you may not use this file except in compliance with the License.
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+* You may obtain a copy of the License at the root of this project source code,
+* or at:
+*     https://centrality.ai/licenses/gplv3.txt
+*     https://centrality.ai/licenses/lgplv3.txt
+*/
+#![cfg_attr(not(feature = "std"), no_std)]
+
+//! # NFT Module
+//!
+//! Provides the basic creation and management of dynamic NFTs (created at runtime).
+//!
+//! Intended to be used "as is" by dapps and provide basic NFT feature set for smart contracts
+//! to extend.
+
+use cennznet_primitives::types::{AssetId, Balance};
```

Comment:
Relying on cennznet_primitives makes the module less flexible plus type collision possibilities in the future where for example asset id is set differently in different places. Most other modules depend on generic types for all their used types. I suggest we follow the same pattern for these two as well.

Reply:
Yeah I realized we pay alot for the generic-ness mostly as a habit from substrate and plug but it doesn't give us any value in CENNZnet, just makes code confusing maintaining all the type conversions

Reply:
I get what you mean, but the mix of concrete types and generic types especially for the types that are defined generic in some other modules, doesn't seem like a good pattern to me. We are specifying asset id and balance in the runtime for some modules and not for this one! This is a possibility for type collission and looks untidy to me.

Reply:
I would be leaning toward agreeing with you for the types that are specific to this module.